python_modules/dagster/dagster_tests/asset_defs_tests/test_decorators.py

```diff
@@ -1159,6 +1159,66 @@ def two_assets(x, y):
     }
 
 
+def test_graph_asset_w_ins():
+    @asset
+    def upstream():
+        return 1
+
+    @op
+    def plus_1(x):
+        return x + 1
+
+    @graph_asset(ins={"one": AssetIn("upstream")})
+    def foo(one):
+        return plus_1(one)
+
+    result = materialize_to_memory([upstream, foo])
+    assert result.success
+    assert result.output_for_node("foo") == 2
+
+    @graph_multi_asset(outs={"first_asset": AssetOut()}, ins={"one": AssetIn("upstream")})
+    def bar(one):
+        x = plus_1(one)
+        return {"first_asset": x}
+
+    result = materialize_to_memory([upstream, bar])
+    assert result.success
+    assert result.output_for_node("bar", "first_asset") == 2
+
+    @asset
+    def another_upstream():
+        return 2
+
+    def concat_op_factory(**kwargs):
+        ins = {i: In() for i in kwargs}
+
+        @op(ins=ins)
+        def concat_op(**kwargs):
+            return list(kwargs.values())
+
+        return concat_op
+
+    @graph_asset(ins={"one": AssetIn("upstream"), "two": AssetIn("another_upstream")})
```

Comment:
This case is fully independent from the case above it, right? If so, separating it out into its own test would make this clearer.